### PR TITLE
feat(search): #MEX-64 search bar in front

### PR DESCRIPTION
**Pull request by sjollois**
## Describe your changes
Add search bar in front to search by title or owner name 
## Checklist tests
search specific world by title and then by author
## Issue ticket number and link
[https://jira.support-ent.fr/browse/MEX-64](url)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qperie — APPROVED** [review]

File: src/main/resources/public/ts/controllers/minetest.controller.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ interface IViewModel {
     getWiki(): string;
     getDownload(): string;
     refreshWorldList(world?: IWorld): any;
+    filterWorlds(searching: string): any;
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/minetest.controller.ts]: typage

File: src/main/resources/public/ts/controllers/minetest.controller.ts [hunk 2]
```diff
@@ -106,6 +107,17 @@ class Controller implements ng.IController, IViewModel {
             self.getWorld(world);
         }
     }
+
+    filterWorlds(searching:string): any {
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/minetest.controller.ts]: typage (World[])

File: src/main/resources/public/ts/controllers/minetest.controller.ts [hunk 3]
```diff
@@ -106,6 +107,17 @@ class Controller implements ng.IController, IViewModel {
...
+        if(!!searching){
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/minetest.controller.ts]: juste if (searching) ce n'est pas suffisant ?

File: src/main/resources/public/ts/controllers/minetest.controller.ts [hunk 4]
```diff
@@ -106,6 +107,17 @@ class Controller implements ng.IController, IViewModel {
...
+            return _.filter(this.worlds.all, function (world) {
+                return world.owner_name.toLowerCase().includes(searching.toLowerCase()) ||
+                    world.title.toLowerCase().includes(searching.toLowerCase());
+            });
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/minetest.controller.ts]:
```suggestion
          return _.filter(this.worlds.all, (world) => 
                world.owner_name.toLowerCase().includes(searching.toLowerCase()) || 
                world.title.toLowerCase().includes(searching.toLowerCase())
            );
```

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**